powersimdata/design/investment/investment_costs.py

```diff
@@ -159,21 +159,24 @@ def get_branch_mult(x, bus_reg, ac_reg_mult, branch_lookup_alerted=set()):
     # otherwise re-run mapping script on mis-matching buses. These buses are missing
     # in region file
     bus = grid_new.bus
-    bus_fix_index = bus[~bus.index.isin(bus_reg.index)].index
-    bus_mask = bus[~bus.index.isin(bus_fix_index)]
-    bus_mask = merge_keep_index(bus_mask, bus_reg, how="left", on="bus_id")
+    mapped_buses = bus.query("index in @bus_reg.index")
+    missing_bus_indices = set(bus.index) - set(bus_reg.index)
+    mapped_buses = merge_keep_index(mapped_buses, bus_reg, how="left", on="bus_id")
     # these buses have incorrect lat/lon values in the region mapping file.
     #   re-running the region mapping script on those buses only.
-    bus_fix_index2 = bus_mask[
-        ~np.isclose(bus_mask.lat_x, bus_mask.lat_y)
-        | ~np.isclose(bus_mask.lon_x, bus_mask.lon_y)
+    misaligned_bus_indices = mapped_buses[
+        ~np.isclose(mapped_buses.lat_x, mapped_buses.lat_y)
+        | ~np.isclose(mapped_buses.lon_x, mapped_buses.lon_y)
```

Comment:
~(lat & lon) ?

Reply:
Is this some sort of pandas magic?

Reply:
I've no idea. Just De Morgan's law?

Reply:
What these lines are doing is comparing whether the `lat` & `lon` from the `bus` dataframe match, within some tolerance, the `lat` & `lon` from the `bus_reg` dataframe (which is loaded from CSV for speed). It identifies any line in which this is not the case.

If you are asking whether `~(np.isclose(...) & np.isclose(...))` would do the same thing, I think the answer is yes.

Reply:
Aha, exactly. Whenever not A or not B, I usually do not (A and B) for cleaner logic (personal preference though). They are equivalent.